cluster.go

```diff
@@ -1813,3 +1741,231 @@ func (c *cluster) setStatic(hosts []string) error {
 	}
 	return nil
 }
+
+type ClusterStatus struct {
+	ClusterID string
+	State     string
+	Nodes     []*Node
+}
+
+type ResizeInstruction struct {
+	JobID         int64
+	Node          *Node
+	Coordinator   *Node
+	Sources       []*ResizeSource
+	Schema        *Schema
+	ClusterStatus *ClusterStatus
+}
+
+func decodeResizeInstruction(ri *internal.ResizeInstruction) ResizeInstruction {
+	return &ResizeInstruction{
+		JobID:         ri.JobID,
+		Node:          DecodeNode(ri.Node),
+		Coordinator:   DecodeNode(ri.Coordinator),
+		Sources:       decodeResizeSources(ri.Sources),
+		Schema:        decodeSchema(ri.Schema),
+		ClusterStatus: decodeClusterStatus(ri.ClusterStatus),
+	}
+}
+
+type ResizeSource struct {
+	Node  *Node  `protobuf:"bytes,1,opt,name=Node" json:"Node,omitempty"`
+	Index string `protobuf:"bytes,2,opt,name=Index,proto3" json:"Index,omitempty"`
+	Field string `protobuf:"bytes,3,opt,name=Field,proto3" json:"Field,omitempty"`
+	View  string `protobuf:"bytes,4,opt,name=View,proto3" json:"View,omitempty"`
+	Shard uint64 `protobuf:"varint,5,opt,name=Shard,proto3" json:"Shard,omitempty"`
+}
+
+func decodeResizeSources(srcs []*internal.ResizeSource) []*ResizeSource {
+	new := make([]*ResizeSource, 0, len(srcs))
+	for _, src := range srcs {
+		new = append(new, decodeResizeSource(src))
+	}
+	return new
+}
+
+func decodeResizeSource(rs *internal.ResizeSource) *ResizeSource {
+	return &ResizeSource{
+		Node:  DecodeNode(rs.Node),
+		Index: rs.Index,
+		Field: rs.Field,
+		View:  rs.View,
+		Shard: rs.Shard,
+	}
+}
+
+// Schema is a schema
+type Schema struct {
+	Indexes []*IndexInfo
+}
+
+func decodeSchema(s *internal.Schema) *Schema {
+	return &Schema{
+		Indexes: decodeIndexes(s.Indexes),
+	}
+}
+
+func decodeIndexes(idxs []*internal.Index) []*IndexInfo {
+	new := make([]*IndexInfo, 0, len(idxs))
+	for _, idx := range idxs {
+		new = append(new, decodeIndex(idx))
+	}
+	return new
+}
+
+func decodeIndex(idx *internal.Index) *IndexInfo {
+	return &IndexInfo{
+		Name:   idx.Name,
+		Fields: decodeFields(idx.Fields),
+	}
+}
+
+func decodeFields(fs []*internal.Field) []*FieldInfo {
+	new := make([]*FieldInfo, 0, len(fs))
+	for _, f := range fs {
+		new = append(new, decodeField(f))
+	}
+	return new
+}
+
+func decodeField(f *internal.Field) *FieldInfo {
+	fi := &FieldInfo{
+		Name:    f.Name,
+		Options: *decodeFieldOptions(f.Meta),
+		Views:   make([]*viewInfo, 0, len(f.Views)),
+	}
+	for _, viewname := range f.Views {
+		fi.Views = append(fi.Views, &viewInfo{Name: viewname})
+	}
+	return fi
+}
+
+// EncodeNodes converts a slice of Nodes into its internal representation.
+func EncodeNodes(a []*Node) []*internal.Node {
+	other := make([]*internal.Node, len(a))
+	for i := range a {
+		other[i] = EncodeNode(a[i])
+	}
+	return other
+}
+
+// EncodeNode converts a Node into its internal representation.
+func EncodeNode(n *Node) *internal.Node {
```

Comment:
Why is this not located in the internal package, or are you having doubts about that strategy?

Reply:
I'm just trying to define matching structs for all the internal structs right now. It's hairier than I thought it was gonna be, but I think I've got it pretty close